### PR TITLE
Expand intro, add sections on methods, purposes, controllers; refactor.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
- Expands the Introduction section, focuses it on the functionality that proofs enable.
- Refactors the Terminology section, extracts `canonicalization algorithm`, `message digest algorithm` and `proof algorithm` to their own Advanced Terminology section.
- Removes `nonce` from Terminology and Overview sections, and the various examples.
- Changes reference from `proof suite` to `proof type`
- Adds a Verification Attributes section, with sub-sections for Proof Purpose, Verification Method and Controller.

(wip)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -139,21 +139,53 @@
...
+
+<ul>
+  <li>
+Make statements that can be shared without loss of trust, because their
+authorship can be verified by a third party (e.g. Verifiable Credentials,
+social media posts, so on).
+  </li>
+  <li>
+Authenticate as the controller of a Decentralized Identifier (DID) to some
+domain.
```
Comment (dlongley) [inline index.html]: You don't authenticate as the controller of a DID but as the DID subject. This statement would only be true conditionally: if the DID subject *happens* to be the same as the controller. It is also too specific, so I recommend:

```
Authenticate as an entity identified by a particular identifier, for example, as the subject identified by a Decentralized Identifier (DID), i.e., a DID subject.
```

**Review by msporny — CHANGES_REQUESTED** [review]
Overall, good direction. Minor nits here and there, content is solid, though.

File: index.html [hunk 2]
```diff
@@ -462,7 +582,6 @@ <h3>Proof Sets</h3>
     "created": "2019-08-23T20:21:34Z",
     "verificationMethod": "did:example:123456#key1",
     "domain": "example.org",
-    "nonce": "2bbgh3dgjg2302d-d2b3gi423d42",
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html:465]: We're still gonna support it, but not encourage it or highlight it in examples. (We will highlight `challenge`, however).

File: index.html [hunk 3]
```diff
@@ -139,21 +139,53 @@
 
     <section>
       <h2>Introduction</h2>
+      <p>
+Cryptographic proofs enable functionality that's extremely useful to
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Thanks, fixed.

File: index.html (hunk 1 above)
Comment (dmitrizagidulin) [reply index.html]: Thanks, changed.

File: index.html [hunk 4]
```diff
@@ -139,21 +139,53 @@
...
+Make statements that can be shared without loss of trust, because their
+authorship can be verified by a third party (e.g. Verifiable Credentials,
+social media posts, so on).
+  </li>
+  <li>
+Authenticate as the controller of a Decentralized Identifier (DID) to some
+domain.
+  </li>
+  <li>
+Invoke capabilities where the invocation can be verified by another party.
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Good point, yeah.

File: index.html [hunk 5]
```diff
@@ -406,14 +447,6 @@ <h2>Linked Data Proof Overview</h2>
 proof or hash algorithms, those must be given via other properties in the
 nodes.</div>
 
-<div class="issue">Add a note indicating that this specification should not
```
Comment (dmitrizagidulin) [inline index.html:409]: Removing this issue, as it no longer applies.

File: index.html [hunk 6]
```diff
@@ -206,21 +238,45 @@ <h2>Terminology</h2>
...
+A cryptographic key that can be used to verify digital proofs created
+with a corresponding <a>private key</a>.
+        </dd>
+        <dt><dfn>private key</dfn></dt>
+        <dd>
+A cryptographic key that can be used to generate digital proofs.
+        </dd>
+        <dt><dfn>proof type</dfn></dt>
+        <dd>
+(Also known as a <dfn>proof suite</dfn>.) A specified set of cryptographic
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Sounds good, will restrict it to "proof type" and "cryptographic suite".

File: index.html [hunk 7]
```diff
@@ -432,12 +474,90 @@ <h2>Linked Data Proof Overview</h2>
...
+  authentication protocol.
+          </dd>
+          <dt>assertionMethod</dt>
+          <dd>
+  Indicates that a proof can only be used for making assertions (for example,
+  signing a Verifiable Credential).
+          </dd>
+          <dt>keyAgreement</dt>
+          <dd>
+  Indicates that a proof is used for the purpose of key agreement, for encryption.
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Reworded, thanks.

File: index.html [hunk 8]
```diff
@@ -139,21 +139,53 @@
...
+      </p>
+
+<ul>
+  <li>
+Make statements that can be shared without loss of trust, because their
+authorship can be verified by a third party (e.g. Verifiable Credentials,
+social media posts, so on).
+  </li>
+  <li>
+Authenticate as the controller of a Decentralized Identifier (DID) to some
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Reworded.

File: index.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Rephrased.

File: index.html [hunk 9]
```diff
@@ -302,8 +351,9 @@ <h2>Linked Data Proof Overview</h2>
       <dl style="margin-left: 1em;">
         <dt>type (required)<dt>
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply index.html]: Fixed.

**Comment by dmitrizagidulin** [comment]
@msporny @dlongley I think I've addressed your comments to the PR. If you don't mind, I'd like to merge this iteration, and work on the next one (refactoring the Algorithm section etc).

**Comment by dlongley** [comment]
@dmitrizagidulin +1 to merge from me.

**Review by dlongley — APPROVED** [review]
Good iteration.

**Pull request merged**